wrappers/js/packages/foundation/utils.cc

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) 2015-2019 Virgil Security, Inc.
+//
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+//    (1) Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//
+//    (2) Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in
+//    the documentation and/or other materials provided with the
+//    distribution.
+//
+//    (3) Neither the name of the copyright holder nor the names of its
+//    contributors may be used to endorse or promote products derived from
+//    this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE AUTHOR ''AS IS'' AND ANY EXPRESS OR
+// IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+// WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT,
+// INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+// (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+// HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+// IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//
+// Lead Maintainer: Virgil Security Inc. <support@virgilsecurity.com>
+
+#include "utils.h"
+
+Nan::MaybeLocal<v8::Object> utils::VirgilBufferToNodeBuffer(vsc_buffer_t* buffer) {
+  size_t buffer_len = vsc_buffer_len(buffer);
+  char* data = new char[buffer_len];
+  std::memcpy(data, vsc_buffer_bytes(buffer), buffer_len);
```

Comment:
!!! This memory copy should be reduced.
Next pattern can be used.
1. Allocate data of max size.
2. Create an object of the class `vsc_buffer_t` without internal bytes `vsc_buffer_new()`.
3. Pass allocated data the `vsc_buffer_t ` object with the method `vsc_buffer_use(...)`.
4. Call wrapped C method and pass the buffer to it.
5. Create class `node::Buffer` with data and an actual length.